packages/client/src/comp/Address/Container.ts

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import { createContainer } from 'unstated-next'
+import { AddressInputParts, Address, getState, isState } from '../../common'
+import { useAppHistory, Path } from '../../lib/path'
+import { AddressContainer } from '../../lib/unstated'
+
+type InputId = keyof AddressInputParts
+
+const removeOptionalFields = (fields: AddressInputParts) => ({
+  ...fields,
+  unit: fields.unit ? fields.unit : undefined,
+})
+
+const handleDefaultState = (rawState: string) => {
+  // check if a state abbreviation is used
+  if (rawState.length === 2) {
+    const state = getState(rawState)
+    if (state) return state
+  }
+  if (isState(rawState)) {
+    return rawState
+  }
+
+  // Return an empty string to prevent errors with invalid states in the application
+  return ''
+}
+
+const handleDefault = (
+  id: InputId,
+  address: Address | null,
+  path: Path | null,
+  query: Record<string, string>,
+): string => {
+  if ((id === 'postcode' || id === 'state') && path?.type === 'address') {
```

Comment:
Let's use `switch` to break this function by `path?.type` (we can have a default that does nothing).

Then we process the logic for each page separately.